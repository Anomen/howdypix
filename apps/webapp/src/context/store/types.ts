import React from "react";

export type StoreContextData = {
  currentAlbum: string | null;
  currentSource: string | null;
  withLayout: boolean;
  rightPanel: React.ReactElement | null;
  setCurrentAlbum: (album: StoreContextData["currentAlbum"]) => void;
  setCurrentSource: (source: StoreContextData["currentSource"]) => void;
  setRightPanel: (panel: StoreContextData["rightPanel"]) => void;
  setWithLayout: (panel: StoreContextData["withLayout"]) => void;
};
