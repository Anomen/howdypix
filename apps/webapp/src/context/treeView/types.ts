import {
  GetTreeAlbums,
  GetTreeSources,
} from "../../__generated__/schema-types";

export type TreeItem = {
  album: string | null;
  source: string;
};

export type TreeItemWithParent = TreeItem & {
  parent: string | null;
};

export type AlbumWithNodeId = GetTreeAlbums & {
  nodeId: string;
};
export type SourceWithNodeId = GetTreeSources & {
  nodeId: string;
};

export type TreeViewContextData = {
  expand: (item: TreeItem) => void;
  collapse: (item: TreeItemWithParent) => void;
  toggle: (item: TreeItemWithParent) => void;
  expandedNodeIds: string[];
  sources: SourceWithNodeId[];
  albums: AlbumWithNodeId[];
};
