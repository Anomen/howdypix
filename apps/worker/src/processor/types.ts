import { ExifData, HFile } from "@howdypix/shared-types";

export type FetchExif = (root: string, path: string) => Promise<ExifData>;
export type CreateThumbnails = (
  thumbnailsDir: string,
  root: string,
  hfile: HFile
) => Promise<string[]>;

export type FileProcessor = {
  fetchExif: FetchExif;
  createThumbnails: CreateThumbnails;
};
