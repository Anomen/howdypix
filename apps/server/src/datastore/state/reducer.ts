import { Reducer } from "redux";
import { uniqWith, isEqual } from "lodash";

type StateSearch = {
  lastUpdatedPhoto: Date | null;
  album: string;
  source: string;
};

type StateAlbum = {
  lastUpdatedPhoto: Date | null;
};

type StateSearchAlbumRelation = {
  searchId: number;
  albumId: number;
};

export type State = {
  searches: {
    [id: number]: StateSearch;
  };
  albums: {
    [id: number]: StateAlbum;
  };
  searchAlbumRelation: StateSearchAlbumRelation[];
};

export type Action =
  | {
      type: "RESET_STORE";
      data: State;
    }
  | {
      type: "UPDATE_SEARCH_LAST_UPDATED_PHOTO";
      searchId: number;
      data: {
        lastUpdatedPhoto: Date;
      };
    }
  | {
      type: "STORE_NEW_SEARCH";
      searchId: number;
      data: StateSearch;
    }
  | {
      type: "UPDATE_ALBUM_LAST_UPDATED_PHOTO";
      albumId: number;
      data: {
        lastUpdatedPhoto: Date;
      };
    }
  | {
      type: "STORE_NEW_ALBUM";
      albumId: number;
      data: StateAlbum;
    }
  | {
      type: "STORE_NEW_SEARCH_ALBUM_RELATION";
      albumId: number;
      searchId: number;
    };

// ========================================================================
// Reducer
// ========================================================================
export const reducer: Reducer<State, Action> = (
  state = { searches: {}, albums: {}, searchAlbumRelation: [] },
  action
) => {
  // console.log(action);
  switch (action.type) {
    case "RESET_STORE":
      return action.data;

    case "UPDATE_SEARCH_LAST_UPDATED_PHOTO": {
      const lastUpdatedPhoto =
        state.searches[action.searchId]?.lastUpdatedPhoto;

      if (
        (lastUpdatedPhoto && action.data.lastUpdatedPhoto > lastUpdatedPhoto) ||
        lastUpdatedPhoto === null
      ) {
        return {
          ...state,
          searches: {
            ...state.searches,
            [action.searchId]: {
              ...state.searches[action.searchId],
              lastUpdatedPhoto: new Date(action.data.lastUpdatedPhoto),
            },
          },
        };
      }

      return state;
    }

    case "UPDATE_ALBUM_LAST_UPDATED_PHOTO": {
      const lastUpdatedPhoto = state.albums[action.albumId]?.lastUpdatedPhoto;

      if (
        (lastUpdatedPhoto && action.data.lastUpdatedPhoto > lastUpdatedPhoto) ||
        lastUpdatedPhoto === null
      ) {
        return {
          ...state,
          albums: {
            ...state.albums,
            [action.albumId]: {
              lastUpdatedPhoto: new Date(action.data.lastUpdatedPhoto),
            },
          },
        };
      }

      return state;
    }

    case "STORE_NEW_ALBUM":
      return {
        ...state,
        albums: {
          ...state.albums,
          [action.albumId]: {
            ...action.data,
            lastUpdatedPhoto: action.data.lastUpdatedPhoto
              ? new Date(action.data.lastUpdatedPhoto)
              : null,
          },
        },
      };

    case "STORE_NEW_SEARCH":
      return {
        ...state,
        searches: {
          ...state.searches,
          [action.searchId]: {
            ...action.data,
            lastUpdatedPhoto: action.data.lastUpdatedPhoto
              ? new Date(action.data.lastUpdatedPhoto)
              : null,
          },
        },
        searchAlbumRelation: uniqWith(
          [
            ...state.searchAlbumRelation,
            {
              searchId: action.searchId,
              albumId: action.data.album,
            },
          ],
          isEqual
        ),
      };

    case "STORE_NEW_SEARCH_ALBUM_RELATION":
      return {
        ...state,
        searchAlbumRelation: [
          ...state.searchAlbumRelation,
          ...(state.searchAlbumRelation.find(
            (relation) =>
              relation.albumId === action.albumId &&
              relation.searchId === action.searchId
          )
            ? [{ albumId: action.albumId, searchId: action.searchId }]
            : []),
        ],
      };

    default:
      return state;
  }
};
