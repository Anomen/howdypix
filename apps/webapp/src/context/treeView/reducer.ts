import { uniq, uniqBy } from "lodash";
import { QueryLazyOptions } from "@apollo/react-hooks";
import {
  GetTreeAlbums,
  GetTreeQuery,
  GetTreeQueryVariables,
  GetTreeSources,
} from "../../__generated__/schema-types";
import {
  AlbumWithNodeId,
  SourceWithNodeId,
  TreeItem,
  TreeItemWithParent,
} from "./types";

// ========================================================================
// Utility functions
// ========================================================================
const selectParents = (
  data: GetTreeQuery["getTree"]["albums"] | null | undefined,
  dir: string
): { [key: string]: boolean } => {
  let ret: { [key: string]: boolean } = {};

  if (!data) {
    ret[dir] = true;
  } else {
    data.forEach((d): void => {
      if (d.dir === dir) {
        ret[d.dir] = true;

        if (d.parentDir) {
          ret = {
            ...ret,
            ...selectParents(data, d.parentDir),
          };
        }
      }
    });
  }

  return ret;
};

const calculateExpandedNodeIds = (
  data: AlbumWithNodeId[],
  selectedItems: { [name: string]: boolean }
): string[] => {
  const ret: string[] = [];

  data.forEach((d): void => {
    if (d.dir && selectedItems[d.dir]) {
      ret.push(d.nodeId);
    }
    if (d.parentDir === "" && selectedItems[d.source]) {
      ret.push(d.source);
    }
  });

  return uniq(ret);
};

// ========================================================================
// Types
// ========================================================================
export type State = {
  fetchedAlbums: AlbumWithNodeId[];
  fetchedSources: SourceWithNodeId[];
  visibleLeaves: {
    [key: string]: boolean;
  };
  expandedNodeIds: string[];
};

export type Actions =
  | {
      type: "DATA_FETCHED";
      albums: GetTreeAlbums[];
      sources: GetTreeSources[];
      variables: {
        album: string;
        source: string;
      };
    }
  | {
      type: "DISPLAY_LEAF";
      album: string;
      source: string;
    }
  | {
      type: "UPDATE_EXPANDED_NODE_IDS";
    }
  | (TreeItem & {
      type: "EXPAND";
      fetchTree: (
        options?: QueryLazyOptions<GetTreeQueryVariables> | undefined
      ) => void;
    })
  | (TreeItemWithParent & {
      type: "COLLAPSE";
    })
  | (TreeItemWithParent & {
      type: "TOGGLE";
      fetchTree: (
        options?: QueryLazyOptions<GetTreeQueryVariables> | undefined
      ) => void;
    });

// ========================================================================
// Reducer
// ========================================================================
export const reducer = (state: State, action: Actions): State => {
  let newState: State = state;

  switch (action.type) {
    case "DATA_FETCHED":
      newState = {
        ...newState,
        fetchedSources: uniqBy(
          [...(newState.fetchedSources ?? []), ...action.sources],
          "name"
        ).map(
          (source): SourceWithNodeId => ({
            ...source,
            nodeId: source.name,
          })
        ),
        fetchedAlbums: uniqBy(
          [...(newState.fetchedAlbums ?? []), ...action.albums],
          "dir"
        ).map((album) => ({
          ...album,
          nodeId: album.source + album.dir,
        })),
      };

      return reducer(newState, {
        type: "DISPLAY_LEAF",
        source: action.variables.source,
        album: action.variables.album,
      });

    case "DISPLAY_LEAF":
      return reducer(
        {
          ...newState,
          visibleLeaves: {
            ...newState.visibleLeaves,
            ...(action.album !== ""
              ? selectParents(newState.fetchedAlbums, action.album)
              : {}),
            [action.source]: true,
          },
        },
        { type: "UPDATE_EXPANDED_NODE_IDS" }
      );

    case "UPDATE_EXPANDED_NODE_IDS":
      return {
        ...newState,
        expandedNodeIds: calculateExpandedNodeIds(
          newState.fetchedAlbums,
          newState.visibleLeaves
        ),
      };

    case "EXPAND":
      if (action.album === null) {
        if (newState.visibleLeaves[action.source] === undefined) {
          action.fetchTree({
            variables: {
              album: "",
              source: action.source,
            },
          });
        }
      } else if (newState.visibleLeaves[action.album] === undefined) {
        action.fetchTree({
          variables: {
            album: action.album ?? "",
            source: action.source,
          },
        });
      }

      // different than original
      return reducer(
        reducer(newState, {
          type: "DISPLAY_LEAF",
          source: action.source,
          album: action.album ?? "",
        }),
        { type: "UPDATE_EXPANDED_NODE_IDS" }
      );

    case "COLLAPSE":
      return reducer(
        {
          ...newState,
          visibleLeaves: {
            ...newState.visibleLeaves,
            ...(action.album !== null
              ? { [action.album]: false }
              : { [action.source]: false }),
            ...(action.parent ? { [action.parent]: true } : {}),
          },
        },
        { type: "UPDATE_EXPANDED_NODE_IDS" }
      );

    case "TOGGLE":
      if (action.album !== null) {
        if (newState.visibleLeaves[action.album]) {
          return reducer(newState, {
            ...action,
            type: "COLLAPSE",
          });
        }

        return reducer(newState, {
          ...action,
          type: "EXPAND",
        });
      }

      if (newState.visibleLeaves[action.source]) {
        return reducer(newState, {
          ...action,
          type: "COLLAPSE",
        });
      }

      return reducer(newState, {
        ...action,
        type: "EXPAND",
      });

    default:
      return newState;
  }
};
