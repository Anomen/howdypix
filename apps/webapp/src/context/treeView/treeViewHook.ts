import { useContext, useEffect } from "react";
import { TreeViewContext } from "./treeViewContext";
import { TreeViewContextData } from "./types";

type Props = {
  album?: string;
  source?: string;
};

export const useTreeView = ({ album, source }: Props): TreeViewContextData => {
  const context = useContext(TreeViewContext);

  useEffect(() => {
    context.expand({
      album: album ?? null,
      source: source ?? ""
    });
  }, [album, source]);

  return context;
};
