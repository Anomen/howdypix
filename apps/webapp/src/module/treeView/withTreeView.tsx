import React, { ComponentType, useEffect } from "react";
import gql from "graphql-tag";
import { useLazyQuery } from "@apollo/react-hooks";
import { appDebug } from "@howdypix/utils";
import { uniq, uniqBy } from "lodash";
import {
  GetTreeQuery,
  GetTreeQueryVariables
} from "../../__generated__/schema-types";
import {
  RawDataAlbums,
  RawDataSource,
  RawDataSources,
  TreeItem,
  TreeItemWithParent,
  TreeViewProvider
} from "./treeViewContext";

const debug = appDebug("withUser");

type DisplayedLeaves = {
  [key: string]: boolean;
};

//= =======================================
// GraphQL queries
//= =======================================
const GET_TREE_VIEW = gql`
  query GetTree($album: String!, $source: String!) {
    getTree(album: $album, source: $source) {
      sources {
        name
      }
      albums {
        dir
        parentDir
        source
      }
    }
  }
`;
// ========================================================================
// Utility functions
// ========================================================================
const selectParents = (
  data: GetTreeQuery["getTree"]["albums"] | null | undefined,
  dir: string
): { [key: string]: boolean } => {
  let ret: { [key: string]: boolean } = {};

  if (!data) {
    ret[dir] = true;
  } else {
    data.forEach((d): void => {
      if (d.dir === dir) {
        ret[d.dir] = true;

        if (d.parentDir) {
          ret = {
            ...ret,
            ...selectParents(data, d.parentDir)
          };
        }
      }
    });
  }

  return ret;
};

const defaultExpanded = (
  data: RawDataAlbums,
  selectedItems: { [name: string]: boolean }
): string[] => {
  const ret: string[] = [];

  data.forEach((d): void => {
    if (d.dir && selectedItems[d.dir]) {
      ret.push(d.nodeId);
    }
    if (d.parentDir === "" && selectedItems[d.source]) {
      ret.push(d.source);
    }
  });

  return uniq(ret);
};

export const withTreeView = <P extends object>(
  WrappedComponent: ComponentType<P>
): ComponentType<P> => (props): React.ReactElement => {
  // ============================================
  // Hooks
  // ============================================
  // State of the Tree component to know what node ids are unfolded
  const [expandedNodeIds, setExpandedNodeIds] = React.useState<string[]>([]);

  // State of the tree fetched from the server
  const [fetchedAlbums, setAlbums] = React.useState<RawDataAlbums>();
  const [fetchedSources, setSources] = React.useState<RawDataSources>();

  // All the leaves in the tree that are currently shown
  const [visibleLeaves, setVisibleLeaves] = React.useState<DisplayedLeaves>({});

  // ============================================
  // GrapQL query callback
  // ============================================
  const [fetchTree, { data, loading, variables }] = useLazyQuery<
    GetTreeQuery,
    GetTreeQueryVariables
  >(GET_TREE_VIEW);

  // ============================================
  // Effects
  // ============================================
  // If the data is fetched, we update the tree state
  useEffect(() => {
    if (data && !loading) {
      setAlbums(
        uniqBy([...(fetchedAlbums ?? []), ...data.getTree.albums], "dir").map(
          album => ({
            ...album,
            nodeId: album.source + album.dir
          })
        )
      );

      setSources(
        data.getTree.sources.map(
          (source): RawDataSource => ({
            ...source,
            nodeId: source.name
          })
        )
      );
    }
  }, [data, loading]);

  // If there is a new tree structure that has been fetched, we update the visible leaves
  useEffect(() => {
    if (fetchedAlbums && fetchedSources) {
      if (variables.album !== "") {
        setVisibleLeaves({
          ...visibleLeaves,
          ...selectParents(fetchedAlbums, variables.album),
          [variables.source]: true
        });
      } else if (variables.album === "") {
        setVisibleLeaves({
          ...visibleLeaves,
          [variables.source]: true
        });
      }
    }
  }, [fetchedAlbums, fetchedSources]);

  // When we change the visible leaves, we update the structure in the UI Tree View
  useEffect(() => {
    if (fetchedAlbums) {
      setExpandedNodeIds(defaultExpanded(fetchedAlbums, visibleLeaves));
    }
  }, [visibleLeaves]);

  const expand = (item: TreeItem): void => {
    if (item.album === null) {
      if (visibleLeaves[item.source] === undefined) {
        fetchTree({
          variables: {
            album: "",
            source: item.source
          }
        });
      }
    } else if (visibleLeaves[item.album] === undefined) {
      fetchTree({
        variables: {
          album: item.album ?? "",
          source: item.source
        }
      });
    }

    setVisibleLeaves({
      ...visibleLeaves,
      ...(item.album ? selectParents(fetchedAlbums, item.album) : {}),
      [item.source]: true
    });
  };

  const collapse = (item: TreeItemWithParent): void => {
    const newVisibleLeaves = {
      ...visibleLeaves,
      [item.source]: true
    };

    if (item.album !== null) {
      newVisibleLeaves[item.album] = false;
    } else {
      newVisibleLeaves[item.source] = false;
    }

    if (item.parent) {
      newVisibleLeaves[item.parent] = true;
    }

    console.log(newVisibleLeaves);

    setVisibleLeaves(newVisibleLeaves);
  };

  const toggle = (item: TreeItemWithParent): void => {
    if (item.album !== null) {
      if (visibleLeaves[item.album]) {
        collapse(item);
      } else {
        expand(item);
      }
    } else if (visibleLeaves[item.source]) {
      collapse(item);
    } else {
      expand(item);
    }
  };

  return (
    <TreeViewProvider
      value={{
        expand,
        collapse,
        toggle,
        expandedNodeIds,
        sources: fetchedSources ?? [],
        albums: fetchedAlbums ?? []
      }}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <WrappedComponent {...props} />
    </TreeViewProvider>
  );
};
