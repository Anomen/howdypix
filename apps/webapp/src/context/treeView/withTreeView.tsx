import React, { ComponentType, Reducer, useEffect, useReducer } from "react";
import gql from "graphql-tag";
import { useLazyQuery } from "@apollo/react-hooks";
import {
  GetTreeQuery,
  GetTreeQueryVariables,
} from "../../__generated__/schema-types";
import { TreeViewProvider } from "./treeViewContext";
import { TreeItem, TreeItemWithParent } from "./types";
import { Actions, reducer, State } from "./reducer";

//= =======================================
// GraphQL queries
//= =======================================
const GET_TREE_VIEW = gql`
  query GetTree($album: String!, $source: String!) {
    getTree(album: $album, source: $source) {
      sources {
        name
        preview
        nbImages
        nbAlbums
      }
      albums {
        dir
        parentDir
        source
        preview
        nbImages
        nbAlbums
      }
    }
  }
`;

export const withTreeView = <P extends object>(
  WrappedComponent: ComponentType<P>
): ComponentType<P> => (props): React.ReactElement => {
  // ============================================
  // Hooks
  // ============================================
  const [state, dispatch] = useReducer<Reducer<State, Actions>>(reducer, {
    fetchedAlbums: [],
    fetchedSources: [],
    visibleLeaves: {},
    expandedNodeIds: [],
  });

  // ============================================
  // GrapQL query callback
  // ============================================
  const [fetchTree, { data, loading, variables }] = useLazyQuery<
    GetTreeQuery,
    GetTreeQueryVariables
  >(GET_TREE_VIEW);

  // ============================================
  // Effects
  // ============================================
  // If the data is fetched, we update the tree state
  useEffect(() => {
    if (data && !loading) {
      dispatch({
        type: "DATA_FETCHED",
        variables,
        albums: data.getTree.albums,
        sources: data.getTree.sources,
      });
    }
  }, [data, loading]);

  const expand = (item: TreeItem): void => {
    dispatch({
      type: "EXPAND",
      ...item,
      fetchTree,
    });
  };

  const collapse = (item: TreeItemWithParent): void => {
    dispatch({
      type: "COLLAPSE",
      ...item,
    });
  };

  const toggle = (item: TreeItemWithParent): void => {
    dispatch({
      type: "TOGGLE",
      ...item,
      fetchTree,
    });
  };

  return (
    <TreeViewProvider
      value={{
        expand,
        collapse,
        toggle,
        expandedNodeIds: state.expandedNodeIds,
        sources: state.fetchedSources,
        albums: state.fetchedAlbums,
      }}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <WrappedComponent {...props} />
    </TreeViewProvider>
  );
};
