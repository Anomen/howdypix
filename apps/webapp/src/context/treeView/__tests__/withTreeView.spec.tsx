import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { withTreeView } from "../withTreeView";
import { useTreeView } from "../treeViewHook";
import { reducer } from "../reducer";
import { TreeViewContextData } from "../types";

jest.mock("../reducer", () => ({
  reducer: jest.fn(() => ({
    fetchedAlbums: [],
    fetchedSources: [],
    visibleLeaves: {},
    expandedNodeIds: []
  }))
}));

const testHook = async (
  callback: (contextData: TreeViewContextData) => void
): Promise<void> => {
  const Component: React.FC<{}> = () => {
    const hook = useTreeView({ album: "album1", source: "source1" });

    return (
      <button type="button" onClick={(): void => callback(hook)}>
        button
      </button>
    );
  };

  const WrappedComponent = withTreeView(Component);

  const { findByText } = render(<WrappedComponent />);

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (reducer as jest.MockedFunction<any>).mockClear();

  (await findByText("button")).click();
};

describe("withTreeView", () => {
  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (reducer as jest.MockedFunction<any>).mockClear();
  });

  test("must call the COLLAPSE type in the reducer.", async () => {
    await testHook(({ collapse }) => {
      collapse({ source: "source2", album: "album2", parent: null });
    });

    expect(reducer).toHaveBeenCalled();
    expect(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (reducer as jest.MockedFunction<any>).mock.calls[0][1]
    ).toMatchSnapshot();
  });

  test("must call the EXPAND type in the reducer.", async () => {
    await testHook(({ expand }) => {
      expand({ source: "source2", album: "album2" });
    });

    expect(reducer).toHaveBeenCalled();
    expect(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (reducer as jest.MockedFunction<any>).mock.calls[0][1]
    ).toMatchSnapshot();
  });

  test("must call the TOGGLE type in the reducer.", async () => {
    await testHook(({ toggle }) => {
      toggle({ source: "source2", album: "album2", parent: null });
    });

    expect(reducer).toHaveBeenCalled();
    expect(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (reducer as jest.MockedFunction<any>).mock.calls[0][1]
    ).toMatchSnapshot();
  });
});
