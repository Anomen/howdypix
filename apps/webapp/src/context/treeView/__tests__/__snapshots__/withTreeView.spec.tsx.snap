// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTreeView must call the COLLAPSE type in the reducer. 1`] = `
Array [
  Object {
    "expandedNodeIds": Array [],
    "fetchedAlbums": Array [],
    "fetchedSources": Array [],
    "visibleLeaves": Object {},
  },
  Object {
    "album": "album2",
    "parent": null,
    "source": "source2",
    "type": "COLLAPSE",
  },
]
`;

exports[`withTreeView must call the EXPAND type in the reducer. 1`] = `
Array [
  Object {
    "expandedNodeIds": Array [],
    "fetchedAlbums": Array [],
    "fetchedSources": Array [],
    "visibleLeaves": Object {},
  },
  Object {
    "album": "album2",
    "fetchTree": [Function],
    "source": "source2",
    "type": "EXPAND",
  },
]
`;

exports[`withTreeView must call the TOGGLE type in the reducer. 1`] = `
Array [
  Object {
    "expandedNodeIds": Array [],
    "fetchedAlbums": Array [],
    "fetchedSources": Array [],
    "visibleLeaves": Object {},
  },
  Object {
    "album": "album2",
    "fetchTree": [Function],
    "parent": null,
    "source": "source2",
    "type": "TOGGLE",
  },
]
`;
